### PR TITLE
Update to be compatible with the v0.13 Crossplane release

### LINKED ISSUE #27: Update to v2 Crossplane packging format
### What problem are you facing?

Crossplane v0.13 introduced a new package manager and dropped support for the old package manager. As part of this change, a new v2 package format was introduced. Old provider packages that do not conform will be unable to be installed with Crossplane v0.13 or later.

### How could the Packet Crossplane Provider help solve your problem?

Update to v2 package format as described in crossplane/crossplane#1756 and the linked PRs.

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Makes a variety of updates, including:
- Rename module to match repository
- Drop support for resource claims
- Drop support for resource classes
- Re-generate managed resource methods
- Build as two separate images (package and controller)
- Add `Provider` meta type in `crossplane.yaml`
- Remove dependency on `crossplane/crossplane` 

Fixes #27 

Tested installation with Crossplane v0.13.0 using image built as `hasheddan/crossplane-provider-equinix-metal`.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplane/crossplane/tree/master/PendingReleaseNotes.md

### CODE REVIEW

**Comment by displague** [comment]
I'm going to guess this is a docker build problem based on local configuration.. (`Docker version 19.03.13, build 4484c46d9d`)

```
$ make 
22:40:58 [ .. ] verify dependencies have expected content
all modules verified
22:41:02 [ OK ] go modules dependencies verified
22:41:02 [ .. ] golangci-lint
22:41:04 [ OK ] golangci-lint
22:41:05 [ .. ] go build linux_amd64
22:41:06 [ OK ] go build linux_amd64
22:41:08 [ .. ] docker build build-c57adf82/crossplane-provider-equinix-metal-amd64
quiet currently not implemented
22:41:09 [FAIL]
make[7]: *** [img.build] Error 1
make[6]: *** [do.build.artifacts.linux_amd64] Error 2
make[5]: *** [build.all] Error 2
make[4]: *** [build] Error 2
make[3]: *** [do.build.image.crossplane-provider-equinix-metal] Error 2
make[2]: *** [do.build.artifacts.linux_amd64] Error 2
make[1]: *** [build.all] Error 2
make: *** [build] Error 2
```

I don't see `quiet` anywhere, but there is a `-q` in `build/`.

**Comment by displague** [comment]
It builds after running `docker buildx uninstall`

**Comment by hasheddan** [comment]
@displague just wondering, if you run with `DOCKER_BUILDKIT=1`, does it build even with `buildx` installed? Also, does this PR look good to you? Looks like I can merge, but was waiting for explicit approval 🙂

File: README.md [hunk 1]
```diff
@@ -39,15 +39,13 @@ helm install crossplane --namespace crossplane-system crossplane-alpha/crossplan
...
+curl -sL https://raw.githubusercontent.com/crossplane/crossplane/release-0.13/install.sh | sh
 ```
 
 ## Install the Equinix Metal Provider
 
 ```console
-kubectl crossplane package install --cluster \
-  --namespace crossplane-system \
-  packethost/crossplane-provider-packet:v0.0.2 provider-packet
+kubectl crossplane install provider equinix/crossplane-provider-equinix-metal
```
Comment (displague) [inline README.md:48]: 👍 
I followed https://github.com/packethost/crossplane-provider-equinix-metal/pull/30/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R34-R36 and ended up with 0.12.0. I needed to `helm repo update` first.

I also removed my existing /usr/local/bin/kubectl-crossplane* files before running the new crossplane cli install command.

Comment (hasheddan) [reply README.md:48]: Likely okay to keep as is here right as these directions would theoretically be for a new user?

Comment (displague) [reply README.md:48]: Seems we'll need to update the Provider install instructions to ProviderConfig (with the new secret format).

The cluster/examples files will need revising.

I couldn't figure out how to install from kind's local images:
```
Events:
  Type     Reason         Age                 From                                 Message
  ----     ------         ----                ----                                 -------
  Warning  UnpackPackage  70s (x32 over 16m)  packages/provider.pkg.crossplane.io  cannot unpack package: failed to fetch package from remote: GET https://index.docker.io/v2/equinix/crossplane-provider-equinix-metal/manifests/v0.1.0: MANIFEST_UNKNOWN: manifest unknown; map[Tag:v0.1.0]
```

I've update packagePullPolicy to Never, but I think it wants a hostname somewhere 🤔

Comment (hasheddan) [reply README.md:48]: This PR does not include updating to ProviderConfig and Provider will continue to work as is. This is expected behavior with the packagePullPolicy. It does not work like the old package manager, so you have to load images into the cache to have a pull policy of Never. Take a look at some of the integration tests for providers in Crossplane org if you would like more details.

**Comment by displague** [comment]
@hasheddan with DOCKER_BUILDKIT=1 it still fails.
```
$ docker buildx install
$ export DOCKER_BUILDKIT=1
$ make
09:31:09 [ .. ] verify dependencies have expected content
all modules verified
09:31:14 [ OK ] go modules dependencies verified
09:31:14 [ .. ] golangci-lint
09:31:17 [ OK ] golangci-lint
09:31:18 [ .. ] go build linux_amd64
09:31:18 [ OK ] go build linux_amd64
09:31:21 [ .. ] docker build build-c57adf82/crossplane-provider-equinix-metal-amd64
quiet currently not implemented
09:31:22 [FAIL]
make[7]: *** [img.build] Error 1
make[6]: *** [do.build.artifacts.linux_amd64] Error 2
make[5]: *** [build.all] Error 2
make[4]: *** [build] Error 2
make[3]: *** [do.build.image.crossplane-provider-equinix-metal] Error 2
make[2]: *** [do.build.artifacts.linux_amd64] Error 2
make[1]: *** [build.all] Error 2
make: *** [build] Error 2
```

**Pull request merged**